### PR TITLE
User auth

### PR DESCRIPTION
**Pull request by NickMo91**
i have added the user authentication, and the bcrypt for safer password, 
you can test that by creating a user and trying to signup.

### CODE REVIEW

**Comment by dtofuah** [comment]
dude i don't know if you just copied and paste the auth code because we don't have a renderTamplate function in our code. You were also redirecting to a /home and we dont have anything like that.  I dont understand what you did. Its not working and in your sigup its not saving the username and first and last name in the database only the password. You lost me on here, please fix it

**Comment by wbobeirne** [comment]
Hey @dtofuah, I'm seeing that he added the renderTemplate function to util/renderTemplate.js in this branch, so that's a part of the pull request.

However, I do see some issues in here, so I'll drop a review in to try and help. But try to do the same as well, it helps if you can drop notes exactly where you see issues in code.

File: router/signup.js [hunk 1]
```diff
@@ -1,12 +1,10 @@
...
 const express = require("express");
 const router = express.Router();
 
 
 router.get("/signup", function(req, res) {
-	res.render("template", {
-		page: "signup",
-		pageArg: null,
-	});
+	renderTemplate(res, "Login", "login");
```
Comment (wbobeirne) [inline router/signup.js]: It looks like the "/signup" route is rendering the "login" page here, it should probably render "signup".

File: views/pages/login.ejs [hunk 2]
```diff
@@ -0,0 +1,46 @@
+<!DOCTYPE html>
+<html>
```
Comment (wbobeirne) [inline views/pages/login.ejs]: Pages shouldn't be full `<html>` pages, they should just have the content that's specific to an individual page. You should move the boilerplate into your `views/template.ejs`

File: app.js [hunk 3]
```diff
@@ -27,8 +30,56 @@ app.post("/signup", function(req, res) {
...
+		where: {
+			username: req.body.username,
+		},
+	})
+	.then(function(user) {
+		if (user) {
+			user.comparePassword(req.body.password).then(function(valid) {
+				if (valid) {
+					req.session.userid = user.get("id");
+					res.redirect("/home");
```
Comment (wbobeirne) [inline app.js]: It looks like you don't have a `/home` route yet, so you probably don't want to redirect there.

File: app.js [hunk 4]
```diff
@@ -27,8 +30,56 @@ app.post("/signup", function(req, res) {
 		email: req.body.email,
 		username: req.body.username,
 		password: req.body.password,
+	})
+	.then(function(user) {
+		req.session.userid = user.id;
+		req.redirect("/home");
```
Comment (wbobeirne) [inline app.js]: Same as the issue below, no `/home` route

**Review by wbobeirne — CHANGES_REQUESTED** [review]

**Comment by dtofuah** [comment]
ok looks good

**Pull request merged**